source/Core/Threads/OperatingModes/DebugMenu.cpp

```diff
@@ -127,11 +146,13 @@ void showDebugMenu(void) {
       return;
     else if (b == BUTTON_F_SHORT) {
       screen++;
-#ifdef HALL_SENSOR
-      screen = screen % 17;
-#else
-      screen = screen % 16;
+#ifndef HALL_SENSOR
```

Comment:
Rather than doing yet more muxing of the screen limiter; we should perhaps just skip in the render pass and have same limit for all

Reply:
That's exactly what I'm doing.
See if hall sensor is disabled, then I skip over it.
And if pinecil v2 is disabled I skip over it.
And at the end there is a single limit that doesn't allow it to go above 17.

Reply:
I mean it's not on the render, it's on the button press that the screen skipping is happening, but that seems better to me, because it's one place where the screen index is updated, and also because all the comparisons to check for a skipped screen don't happen on every single frame, but only on a button press, so it's less work for the CPU too.

Reply:
See above comment; but TL;DR I'm just worried that this #ifdef is going to get messy and might be better to just skip inline in the switchcase. Then its only ever a hard coded skip or a draw.

Reply:
see above comment.